test/integration/targets/azure_rm_resource/tasks/main.yml

```diff
@@ -136,12 +136,39 @@
   assert:
     that: keys['response']['keys'][0]['value'] | length > 0
 
-- name: Delete storage - without API version
+- name: Delete unexisting storage account - without API version
   azure_rm_resource:
     polling_timeout: 600
     polling_interval: 60
-    method: DELETE
+    resource_group: "{{ resource_group }}"
+    provider: Storage
+    resource_type: storageAccounts
+    resource_name: justunexisitingname
+    state: absent
+  register: output
+
+- debug:
+    var: output
+
+- name: Assert that no change was detected
```

Comment:
shouldn't the delete call fail?

Reply:
if resource doesn't exist already and state is set to absent, it shouldn't fail.